### PR TITLE
NAS-105922 / 12.0 / Nas 105922

### PR DESCRIPTION
**Pull request by dkmullen**
Adds a one-time dialog to users/groups lists explaining that builtins are hidden; adds mat-tooltip to gear icon; fixes some old references to bsd_user, etc.; removes redundant user and group queries in user-list component. Didn't change 'builtins' to 'system', didn't use filtered query due to a delay in receiving user preferences.

### CODE REVIEW

**Comment by erinix** [comment]
could you please review the text @freenas/docs-team

File: src/app/helptext/account/group-list.ts [hunk 1]
```diff
@@ -9,10 +9,19 @@ group_list_actions_label_delete: T("Delete"),
...
 group_list_dialog_message: T('Delete all users with this primary group?'),
 
-    builtins_dialog: {
-        title: T(' Builtin Groups'),
-        message: T(' builtin groups (default setting is <i>hidden</i>).'),
-        show: T('Show'),
-        hide: T('Hide')
-    }
+builtins_dialog: {
+    title: T(' Builtin Groups'),
```
Comment (Mrt134) [inline src/app/helptext/account/group-list.ts:13]:
```suggestion
    title: T(' Built-in Groups'),
```

File: src/app/helptext/account/group-list.ts [hunk 2]
```diff
@@ -9,10 +9,19 @@ group_list_actions_label_delete: T("Delete"),
...
+    message: T(' builtin groups (default setting is <i>hidden</i>).'),
```
Comment (Mrt134) [inline src/app/helptext/account/group-list.ts:14]:
```suggestion
    message: T(' built-in groups (default setting is <i>hidden</i>).'),
```

File: src/app/helptext/account/group-list.ts [hunk 3]
```diff
@@ -9,10 +9,19 @@ group_list_actions_label_delete: T("Delete"),
...
+    show: T('Show'),
+    hide: T('Hide')
+},
+
+builtinMessageDialog: {
+    title: T('Display Note'),
+    message: T('All builtin groups are \
```
Comment (Mrt134) [inline src/app/helptext/account/group-list.ts:21]:
```suggestion
    message: T('All built-in groups are \
```

File: src/app/helptext/account/group-list.ts [hunk 4]
```diff
@@ -9,10 +9,19 @@ group_list_actions_label_delete: T("Delete"),
...
+ hidden by default. Use the gear icon (top-right) to toggle the display of builtin groups.'),
```
Comment (Mrt134) [inline src/app/helptext/account/group-list.ts:22]:
```suggestion
 hidden by default. Use the gear icon (top-right) to toggle the display of built-in groups.'),
```

File: src/app/helptext/account/group-list.ts [hunk 5]
```diff
@@ -9,10 +9,19 @@ group_list_actions_label_delete: T("Delete"),
...
+    button: T('Close')
+},
+globalConfigTooltip: T('Toggle builtin groups')
```
Comment (Mrt134) [inline src/app/helptext/account/group-list.ts:25]:
```suggestion
globalConfigTooltip: T('Toggle built-in groups')
```

File: src/app/helptext/account/user-list.ts [hunk 6]
```diff
@@ -12,5 +12,13 @@ builtins_dialog: {
     message: T(' builtin users (default setting is <i>hidden</i>).'),
     show: T('Show'),
     hide: T('Hide')
-}
+},
+builtinMessageDialog: {
+    title: T('Display Note'),
+    message: T('All builtin users except <i>root</i> are \
+ hidden by default. Use the gear icon (top-right) to toggle the display of builtin users.'),
```
Comment (Mrt134) [inline src/app/helptext/account/user-list.ts:19]:
```suggestion
    message: T('All built-in users except <i>root</i> are \
 hidden by default. Use the gear icon (top-right) to toggle the display of built-in users.'),
```

File: src/app/helptext/account/user-list.ts [hunk 7]
```diff
@@ -12,5 +12,13 @@ builtins_dialog: {
...
+    button: T('Close')
+},
+globalConfigTooltip: T('Toggle builtin users')
```
Comment (Mrt134) [inline src/app/helptext/account/user-list.ts:22]:
```suggestion
globalConfigTooltip: T('Toggle built-in users')
```

File: src/app/pages/account/groups/group-list/group-list.component.ts [hunk 8]
```diff
@@ -154,6 +162,13 @@ export class GroupListComponent {
          }
       })
   }
+
+  showOneTimeBuiltinMsg() {
+    this.prefService.preferences.showGroupListMessage = false;
+    this.prefService.savePreferences();
+    this.dialogService.confirm(helptext.builtinMessageDialog.title, helptext.builtinMessageDialog.message, 
+      true, helptext.builtinMessageDialog.button, false, '', '', '', '', true);
+  }
```
Comment (erinix) [inline src/app/pages/account/groups/group-list/group-list.component.ts:171]: the dialog should not have a cancel button, and should only be closable by clicking the 'ok' button.  I believe some of these options for dialog service do that

**Pull request merged**